### PR TITLE
Featureimage/new snmpv3 encryption

### PR DESCRIPTION
**Pull request by stark-dev**

### CODE REVIEW

File: lib/include/secw_snmpv3.h [hunk 1]
```diff
@@ -50,6 +53,8 @@ enum Snmpv3PrivProtocol : uint8_t
 {
     DES = 0,
     AES,
+    AES256,
+    AES192 = 4,
```
Comment (etnKlendathu) [inline lib/include/secw_snmpv3.h]: why 4? Any conflicts? And natural order seems be better... AES192, AES256

File: lib/tests/producer_accessor.cpp [hunk 2]
```diff
@@ -500,6 +500,177 @@ void secwProducerAccessorTest(fty::SocketSyncClient& syncClient, mlm::MlmStreamC
         }
     }
 
+    // test 5.7 => insertNewDocument SNMPV3 - SHA256 / AES256
```
Comment (etnKlendathu) [inline lib/tests/producer_accessor.cpp:503]: You could create function or lamda to create document, do not repeat this "footcloth" of exceptions handling.

**Review by etnKlendathu — CHANGES_REQUESTED** [review]

File: lib/include/secw_snmpv3.h (hunk 1 above)
Comment (stark-dev) [reply lib/include/secw_snmpv3.h]: was requested as requirement to have this order

File: lib/tests/producer_accessor.cpp (hunk 2 above)
Comment (stark-dev) [reply lib/tests/producer_accessor.cpp:503]: we should probably review all the UTs and update them to catch2 style

**Review by etnKlendathu — APPROVED** [review]

**Comment by stark-dev** [comment]
Need to recreate the FI branch

**Pull request closed without merging**